app/questionnaire/questionnaire_store_updater.py

```diff
@@ -298,16 +325,32 @@ def update_answers(
 
             answer_updated = self._update_answer(answer_id, list_item_id, answer_value)
             if answer_updated:
-                self._capture_dependencies_for_answer(answer_id)
+                self._capture_section_dependencies_for_answer(answer_id)
+                self._capture_block_dependencies_for_answer(answer_id)
 
-    def update_progress_for_dependant_sections(self) -> None:
-        """Removes dependent blocks from the progress store and updates the progress to IN_PROGRESS.
+    def update_progress_for_dependent_sections(self) -> None:
+        """Section progress is not updated for the current location as it is handled by `handle_post` on block handlers. """
 
-        Section progress is not updated for the current location as it is handled by `handle_post` on block handlers.
+        self._remove_dependent_blocks_from_progress_store()
+
+        for section in self.dependent_sections:
+            is_path_complete = section.is_complete
+
+            if section.is_complete is None:
+                routing_path = self._router.routing_path(
+                    section.section_id, list_item_id=section.list_id
+                )
+                is_path_complete = self._router.is_path_complete(routing_path)
+
+            self.update_section_status(
+                is_complete=is_path_complete,
+                section_id=section.section_id,
+                list_item_id=section.list_id,
+            )
+
+    def _remove_dependent_blocks_from_progress_store(self) -> None:
+        """Removes dependent blocks from the progress store."""
```

Comment:
This method does more than what the name and doc strings says. Based on how you have named `_capture_section_dependencies_for_answer`, this should be something like `_remove_dependent_blocks_and_capture_dependent_sections`

Reply:
Ok, renamed now, thanks!